rwengine/src/audio/EffectSlot.hpp

```diff
@@ -0,0 +1,62 @@
+#ifndef EFFECTSLOT_H
+#define EFFECTSLOT_H
+
+#include <al.h>
+
+#include <memory>
+
+class SoundEffect;
+
+/**
+ * Effect slot.
+ *
+ * Many sound sources can be attached to one slot.
+ * Different effects can be binded to this (e.g reverb, delay).
+ */
+class EffectSlot {
```

Comment:
I have problems with understanding connection between this class and `SoundEffect`. I mean where is `EffectSlot` created, used, deleted. Because instances of `EffectSlot` and `SoundEffect` are in relation 1 to 1. I think solution when one of them contains/owns instances of second would be good enough.